python/lsst/source/injection/inject_engine.py

```diff
@@ -547,6 +592,52 @@ def inject_galsim_objects_into_exposure(
                         err,
                     )
                 continue
+
+            var_template = image_template.copy()
+            var_template *= variance_scale[object_common_bounds]
+
+            if add_noise:
+                # Treat var_template as the Poisson mean of the number of
+                # photons collected by each pixel.
+                noise_template = var_template.copy()
+                negative_variance = var_template.array < 0
+                nonfinite_variance = ~np.isfinite(var_template.array)
+                if np.any(negative_variance):
+                    if logger:
+                        logger.debug("Setting negative-variance pixels to 0 variance for noise generation.")
+                    noise_template.array[negative_variance] = 0
+                if np.any(nonfinite_variance):
+                    if logger:
+                        logger.debug("Setting non-finite-variance pixels to 0 variance for noise generation.")
+                    noise_template.array[nonfinite_variance] = 0
+
+                rng = galsim.BaseDeviate(noise_seed)
+
+                if gains is None:
```

Comment:
@jjbuchanan , is the if/else branch here entirely driven by the non-existence of `galsim.VariablePoissonNoise` ?  I'm wondering if we should simplify and eliminate the branch.  Right now it looks like our choices are:
- Use Gaussian approx to Poisson if individual detector gains are available in the metadata
- If not available, then infer a single gain for the entire image and use Poisson stats with that.
My guess is in either case we probably don't _really_ know the gain well enough to tell the difference between Poisson noise and its Gaussian approximation, but curious to hear what you think.

Reply:
My 2 cents on this: if the purpose of the `if gains is None` check is simply to determine if we're looking at a calexp or a coadd, I would probably favor pulling that if check out into its own conditional and outputting into a named variable that makes this clear. E.g.:
```
is_calexp = True if gains else False
```

Reply:
Makes sense.